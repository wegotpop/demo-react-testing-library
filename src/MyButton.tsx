import * as React from "react";
import { getJSON } from "./popsss";

export function MyButton({
  onClick,
}: {
  onClick: React.MouseEventHandler<HTMLButtonElement>;
}) {
  React.useEffect(() => {
    console.log(getJSON("url"));
  }, []);

  return (
    <div>
      <button
        data-testid="button-to-be-clicked"
        onClick={onClick}
        // onClick={() => {
        //   console.log("i was called");
        // }}
      >
        Click me!
      </button>
    </div>
  );
}
