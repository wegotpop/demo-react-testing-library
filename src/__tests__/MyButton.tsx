import * as React from "react";
import { render, screen } from "@testing-library/react";
import { MyButton } from "../MyButton";

jest.mock("../popsss", () => {
  return { getJSON: (url: string) => ({ other: "value" }) };
});

describe("MuButton tests", () => {
  test("renders learn react link", () => {
    // Create a mock function, which we will then use to check things are being called
    const mockOnClickHandler = jest.fn();

    const result = render(<MyButton onClick={mockOnClickHandler} />);

    // find the element, based on its text content
    // const button = screen.getByText(/Click Me!/i);

    // Find the elelement based on a data-testid attribute
    const button = screen.getByTestId("button-to-be-clicked");

    expect(button).toBeInTheDocument();

    // Click the button
    button.click();
    button.click();

    // Check the mock function was called exactly once
    expect(mockOnClickHandler).toBeCalledTimes(2);
  });
});
