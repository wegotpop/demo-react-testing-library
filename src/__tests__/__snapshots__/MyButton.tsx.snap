// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuButton tests a snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      data-testid="button-to-be-clicked"
    >
      Click me!
    </button>
  </div>
</DocumentFragment>
`;
